chain-registry/src/utils.rs

```diff
@@ -10,48 +10,49 @@ pub const REF: &str = "ffdfbff3a21c7d2dbaac6f1e4730f47063f4bd72";
 
 pub const TEST_CHAINS: &[&str] = &["cosmoshub", "juno", "evmos", "osmosis", "regen"];
 
-pub trait FileName {
+#[async_trait]
+pub trait Fetchable
+where
+    Self: DeserializeOwned,
+{
     fn file_name() -> String;
-}
 
-pub async fn fetch_data<T: FileName>(chain_name: String) -> Result<String, RegistryError> {
-    let url = Builder::new()
-        .scheme(PROTOCOL)
-        .authority(HOST)
-        .path_and_query(
-            format!(
-                "{}/{}/{}/{}",
-                REGISTRY_PATH,
-                REF,
-                chain_name,
-                T::file_name()
+    async fn fetch_data(chain_name: String) -> Result<String, RegistryError> {
```

Comment:
I don't see much reason to split the fetching logic between two different trait methods; it might not make sense for some future use-case of this trait. It is a bit clunky, but I think the logic of `fetch_data` should be folded into `fetch`.